src/Microsoft.AspNet.Hosting.WindowsService/Program.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.AspNet.Hosting.Internal;
+using Microsoft.Extensions.Configuration;
+using System.ServiceProcess;
+
+namespace Microsoft.AspNet.Hosting.WindowsService
+{
+    public class Program : ServiceBase
+    {
+        private const string HostingJsonFile = "hosting.json";
+        private const string EnvironmentVariablesPrefix = "ASPNET_";
+        private const string ConfigFileKey = "config";
+        private IApplication _application;
+
+        public static void Main(string[] args)
+        {
+            Run(new Program());
+        }
+
+        protected override void OnStart(string[] args)
+        {
+            // Allow the location of the json file to be specified via a --config command line arg
+            var tempConfigBuilder = new ConfigurationBuilder().AddCommandLine(args);
+            var tempConfig = tempConfigBuilder.Build();
+            var configFilePath = tempConfig[ConfigFileKey] ?? HostingJsonFile;
+            var configBuilder = CreateHostingConfigurationBuilder(configFilePath, args);
+            var config = configBuilder.Build();
+            var hostBuilder = new WebHostBuilder(config, captureStartupErrors: true);
+            var host = hostBuilder.Build();
+            _application = host.Start();
+        }
+
+        protected override void OnStop()
+        {
+            _application?.Dispose();
+        }
+
+        protected IConfigurationBuilder CreateHostingConfigurationBuilder(string configJsonPath, string[] args)
+        {
+            // We are adding all environment variables first and then adding the ASPNET_ ones
+            // with the prefix removed to unify with the command line and config file formats
+            return new ConfigurationBuilder()
+                .AddJsonFile(configJsonPath, optional: true)
+                .AddEnvironmentVariables()
+                .AddEnvironmentVariables(prefix: EnvironmentVariablesPrefix)
```

Comment:
Yes. The "binPath=" CLI option will have it's options passed in. So when you register the service the "binPath=" parameter can take arguments after the path to the executable/batch file.